azurelinuxagent/daemon/env.py

```diff
@@ -100,6 +107,33 @@ def handle_dhclient_restart(self):
            self.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+    def handle_fstab_update(self):
+        """
+        Look for changes in fstab, mount if the fstab has new modified stamp 
+        If error, only re-check every minute to give an admin
+         time to correct it and avoid filling the log
+        """
+        fstabCurrentModifiedStamp = os.path.getmtime("/etc/fstab")
+        if fstabCurrentModifiedStamp != self.fstabModTime and \
+                            datetime.datetime.now() >  \
+                            self.lastNotice + datetime.timedelta(seconds=60):
+            ret, output = shellutil.run_get_output("mount -av")
```

Comment:
I have been thinking about this, and perhaps we should explore other options. For example, if an entry in fstab is bad, it only matters if it does not fail the boot process. Perhaps we should be monitoring for things like 'nobootwait' in Ubuntu, or the equivalents, to prevent boot issues.